src/tests/end-to-end/tests/details-view/tabstop.test.ts

```diff
@@ -0,0 +1,81 @@
+import { Page } from '../../common/page-controllers/page';
+import { TargetPage } from '../../common/page-controllers/target-page';
+import { Browser } from './../../common/browser';
+import { DEFAULT_PAGE_ELEMENT_WAIT_TIMEOUT_MS } from './../../common/timeouts';
+
+describe('Tabstop tests', () => {
+    describe('tabstop from fastpass', () => {
+        let browser: Browser;
+        let targetPage: TargetPage;
+        let fastPassPage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetPage = await browser.newTargetPage();
+            fastPassPage = await gotoFastPass(browser, targetPage.tabId);
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+            }
+        });
+
+        test('if visualHelper is turned on after starting tabstop and pressing tabs on target page', async () => {
+            const toggleAriaLabel = 'Tab stops';
+
+            await goToTabStopTest();
+            await enableToggleByAriaLabel(toggleAriaLabel);
+
+            await targetPage.bringToFront();
+
+            // press tabs 3 times. todo: DRY this code
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+            await targetPage.keyPress('Tab');
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
+        });
+
+        async function gotoFastPass(browserLocal: Browser, targetTabId: number): Promise<Page> {
+            const popupPage = await browserLocal.newPopupPage(targetPage);
+            let fastPass: Page;
+            await Promise.all([
+                browserLocal.waitForDetailsViewPage(targetPage).then(page => (fastPass = page)),
+                popupPage.clickSelector(popupPageElementIdentifiers.fastPass),
+            ]);
+
+            return fastPass;
+        }
+
+        async function goToTabStopTest(): Promise<void> {
+            await fastPassPage.clickSelector(fastPassSelectors.tabstopNavButtonSelector);
+            // await fastPassPage.disableAnimations();
+            await verifyTabStopPageLoaded();
+        }
+
+        async function enableToggleByAriaLabel(ariaLabel: string): Promise<void> {
+            const toggleSelector = `button[aria-label="${ariaLabel}"]`;
+            const enabledToggleSelector = `${toggleSelector}[aria-checked=true]`;
+            const disabledToggleSelector = `${toggleSelector}[aria-checked=false]`;
+
+            await fastPassPage.waitForDuration(6000);
+            await fastPassPage.clickSelector(disabledToggleSelector);
+
+            const EXTRA_TOGGLE_OPERATION_TIMEOUT_MS = 5000;
+
+            await fastPassPage.waitForSelector(enabledToggleSelector, {
+                timeout: DEFAULT_PAGE_ELEMENT_WAIT_TIMEOUT_MS + EXTRA_TOGGLE_OPERATION_TIMEOUT_MS,
+            });
+        }
+
+        async function verifyTabStopPageLoaded(): Promise<void> {
```

Comment:
should we rename this to `waitFor...` instead of using `verify`??? On a test context, the latter kinds of implies some sort of more explicit assertion of some thing we want as a result or side effect of the test itself, but in this case, this is seems to be use on the **Arrange** part of the test not on the **Assert** part.